### PR TITLE
Change: Refactor Domain creation and disable submission until required fields are present

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

The task here was to disable the create button until the required fields were filled in. Since required fields differ based on what kind of Domain you're creating, I added some conditional logic that felt yucky.

Then I thought, It's Christmas, Jared. Why not do it the hard way? So I did. This PR:

- Refactors the create domain modal to use Formik
- Updates Domains types and tests to be accurate (we were using `Partial<Domain>` for both create and update payloads)
- Does a lot of fancy schmancy stuff with the live validation

## Note to Reviewers

Please make sure I didn't break anything. Creating primary and secondary Domains, with and without tags or default records, should all work as before, including error states.

### CODE REVIEW

File: packages/api-v4/src/domains/types.ts [hunk 1]
```diff
@@ -47,5 +47,24 @@ export interface DomainRecord {
...
 }
 
-export type CreateDomainPayload = Partial<Domain>;
-export type UpdateDomainPayload = Partial<Domain>;
+export interface CreateDomainPayload {
+  domain: string;
+  type: DomainType;
+  master_ips?: string[];
+  soa_email?: string;
+  tags?: string[];
```
Comment (Jskobos) [inline packages/api-v4/src/domains/types.ts:55]: I suspect a lot of the fields in the Update payload are usable here too, though we don't use them. Haven't had a chance to verify all of this through the API yet.

**Review by DevDW — APPROVED** [review]
This is working well. However, I was able to go to create a Secondary Domain and the Create button was enabled even though I didn't provide an IP Address:

Screen Shot 2020-12-23 at 4 03 14 PM

**Review by tiffwong — COMMENTED** [review]
The functionality looks good to me!

The only issue I noticed is that the create button is never disabled on load since there's always a default option selected for primary vs secondary. We should probably make the create button disabled until there's something in all the required fields

**Comment by Jskobos** [comment]
> The functionality looks good to me!
> 
> The only issue I noticed is that the create button is never disabled on load since there's always a default option selected for primary vs secondary. We should probably make the create button disabled until there's something in all the required fields

@tiffwong I'm not able to reproduce this, the button is disabled on load no matter what I do or where I come from. Would you mind checking to see if this still happens for you with the latest changes?

**Review by tiffwong — APPROVED** [review]

**Pull request merged**